### PR TITLE
new features

### PR DESCRIPTION
**Pull request by r-t-a**
- added chromie
- added searchable feature
- updated icons to work on different resolutions
- began work on more test framework stuff
- duplicate espresso dependencies removed

### CODE REVIEW

File: app/src/main/java/com/ryan/heroestopbuilds/MainActivity.java [hunk 1]
```diff
@@ -212,18 +252,31 @@ public String onChildPress(String name) {
...
+        ArrayList<String> finalList = new ArrayList<>();
         String lgSpacing = String.format("%" + 3 + "s", "");
         String smSpacing = String.format("%" + 1 + "s", "");
-        ArrayList<String>finalList = new ArrayList<>();
-        for(int i = 0; i <= levelMod; ++i) {
-            if(i <= 2) {
-                finalList.add("Level " + (1 + 3 * i) + ": " + lgSpacing + popularSkills.get(i));
-            } else {
-                finalList.add("Level " + (1 + 3 * i) + ": " + smSpacing + popularSkills.get(i));
+        //Check first skills, this means we got a Chromie
```
Comment (lewisric) [inline app/src/main/java/com/ryan/heroestopbuilds/MainActivity.java]: This block can be simplified below for instance if one of those OR clauses evals correctly have a var be set to 1 and then add it within your string comprehension, would remove the need for an if() else() with mostly identical code

int isChromie = 0;
if(popularSkills.contains("CompoundingAether") || popularSkills.contains("DeepBreath")
               || popularSkills.contains("TimewalkersPursuit") || popularSkills.contains("PeerIntoTheFuture")) {
isChromie =1
}

//for loop
finalList.add("Level " + (isChromie + 3 * i) + ": " + lgSpacing + popularSkills.get(i));

**Comment by pgicking** [comment]
meme magic is real

File: app/src/main/java/com/ryan/heroestopbuilds/MainActivity.java (hunk 1 above)
Comment (lewisric) [inline app/src/main/java/com/ryan/heroestopbuilds/MainActivity.java]: Lets fuck this shit up, FAM.

Seeing that popularSkills is an ArrayList we can lean on pattern matching the strings within. This'll blow your fuckin mind.

for(String s : popularSkills) {
if(s.matches("(?i)(CompoundingAether | DeepBreath | TimewalkersPursuit | PeerIntoTheFuture)")){
isChromie_? = 1
}
}

I would assume those skill names can be shortened if no other heroes have matching skill names or combos of words

File: app/src/main/java/com/ryan/heroestopbuilds/MainActivity.java [hunk 2]
```diff
@@ -255,25 +255,18 @@ public String prettyPrinter(ArrayList<String> popularSkills) {
         ArrayList<String> finalList = new ArrayList<>();
         String lgSpacing = String.format("%" + 3 + "s", "");
         String smSpacing = String.format("%" + 1 + "s", "");
+        int weGotAChromie = 0;
         //Check first skills, this means we got a Chromie
         if(popularSkills.contains("CompoundingAether") || popularSkills.contains("DeepBreath")
```
Comment (lewisric) [inline app/src/main/java/com/ryan/heroestopbuilds/MainActivity.java]: Lets fuck this shit up, FAM.

Seeing that popularSkills is an ArrayList we can lean on pattern matching the strings within. This'll blow your fuckin mind.

for(String s : popularSkills) {
if(s.matches("(?i)(CompoundingAether|DeepBreath|TimewalkersPursuit|PeerIntoTheFuture)")){
isChromie_? = 1
}
}

I would assume those skill names can be shortened if no other heroes have matching skill names or combos of words

**Comment by lewisric** [comment]
Shit
commented on old commit

File: app/src/main/java/com/ryan/heroestopbuilds/MainActivity.java [hunk 3]
```diff
@@ -212,18 +252,24 @@ public String onChildPress(String name) {
...
+        ArrayList<String> finalList = new ArrayList<>();
         String lgSpacing = String.format("%" + 3 + "s", "");
         String smSpacing = String.format("%" + 1 + "s", "");
-        ArrayList<String>finalList = new ArrayList<>();
-        for(int i = 0; i <= levelMod; ++i) {
-            if(i <= 2) {
-                finalList.add("Level " + (1 + 3 * i) + ": " + lgSpacing + popularSkills.get(i));
+        int weGotAChromie = 0;
+        //Check first skills, this means we got a Chromie
+        if(popularSkills.contains("CompoundingAether") || popularSkills.contains("DeepBreath")
```
Comment (lewisric) [inline app/src/main/java/com/ryan/heroestopbuilds/MainActivity.java]: Lets fuck this shit up, FAM.

Seeing that popularSkills is an ArrayList we can lean on pattern matching the strings within. This'll blow your fuckin mind.

for(String s : popularSkills) {
if(s.matches("(?i)(CompoundingAether|DeepBreath|TimewalkersPursuit|PeerIntoTheFuture)")){
isChromie_? = 1
}
}

I would assume those skill names can be shortened if no other heroes have matching skill names or combos of words

**Comment by r-t-a** [comment]
just making sure travis is ok before pushing the pattern match up

**Comment by lewisric** [comment]
You might need to play with it a bit, should be straight forward though

File: app/src/main/java/com/ryan/heroestopbuilds/MainActivity.java [hunk 4]
```diff
@@ -257,9 +257,10 @@ public String prettyPrinter(ArrayList<String> popularSkills) {
         String smSpacing = String.format("%" + 1 + "s", "");
         int weGotAChromie = 0;
         //Check first skills, this means we got a Chromie
-        if(popularSkills.contains("CompoundingAether") || popularSkills.contains("DeepBreath")
-                || popularSkills.contains("TimewalkersPursuit") || popularSkills.contains("PeerIntoTheFuture")) {
-            weGotAChromie = 1;
+        for(String s : popularSkills) {
+            if (s.matches("(?i)(CompoundingAether|DeepBreath|TimewalkersPursuit|PeerIntoTheFuture)")) {
+                weGotAChromie = 1;
```
Comment (lewisric) [inline app/src/main/java/com/ryan/heroestopbuilds/MainActivity.java]: break; 
After this line, if anything matches there is no need to keep looping. Should save cycles.

**Pull request merged**